packages/backend-common/src/database/DatabaseManager.ts

```diff
@@ -292,13 +299,17 @@ export class DatabaseManager {
   private getConfigForPlugin(pluginId: string): Knex.Config {
     const { client } = this.getClientType(pluginId);
     const role = this.getRoleConfig(pluginId);
+    const auth = this.getAuthConfig(pluginId);
 
-    return {
+    const configForPlugin = {
       ...this.getAdditionalKnexConfig(pluginId),
       client,
       connection: this.getConnectionConfig(pluginId),
       ...(role && { role }),
+      ...(auth && { auth }),
```

Comment:
Hm this is odd. You're adding an `auth` field which, unless I'm misreading things, isn't actually in the `Knex.Config` type. Then you later deconstruct it and do something completely different. 🤔

As a matter of fact, I don't think that you want to do your core work inside the `mergeDatabaseConfig` at all. Could you make it so that all relevant work happens inside `connectors/postgres.ts` instead? It will have the right context for being able to extract the configuration in a type safe way. Then I'd be fine if all connectors _always_ returned the callback form of the knex config, whether necessary or not, and then they can perform config merging inside that callback if they absolutely have to etc.

Does that sound reasonable? Might take a little refactor but the code is spread out in an unfortunate way right now I think

Reply:
I think you reading it incorrectly. I have no intention of using `Knex.Config` there. My idea for `auth` key is to hold authentication credentials like AzureAD. 

>  I don't think that you want to do your core work inside the mergeDatabaseConfig at all. Could you make it so that all relevant work happens inside `connectors/postgres.ts` instead? ....

I do not clearly understand your suggestion, sorry :(
But let's do it step-by-step. I did move code to `connectors/postgres.ts`. Hope it looks better.

What is next?
Can you maybe elaborate a bit on what you have in mind?